cmake/BLTMacros.cmake

```diff
@@ -147,7 +147,8 @@ endmacro(blt_add_target_link_flags)
 ##------------------------------------------------------------------------------
 ## blt_register_library( NAME <libname>
 ##                       DEPENDS_ON [dep1 [dep2 ...]]
-##                       INCLUDES [include1 [include2 ...]] 
+##                       INCLUDES [include1 [include2 ...]]
+##                       TREAT_INCLUDES_AS_SYSTEM [ON] 
```

Comment:
In this case, does it make sense to have ``TREAT_INCLUDES_AS_SYSTEM`` be a cmake ``option``, since users are manually setting it, and the only supported case in for this parameter to be set to ``ON``?

(I know that macro options can be problematic, in general)

Reply:
im trying to avoid options at this point..

Reply:
Here is the relevant code:

```cmake
  if( arg_TREAT_INCLUDES_AS_SYSTEM )
    set(BLT_${uppercase_name}_TREAT_INCLUDES_AS_SYSTEM ON CACHE BOOL "" FORCE)
    mark_as_advanced(BLT_${uppercase_name}_TREAT_INCLUDES_AS_SYSTEM)
  endif()
```
We are currently treating this ``singleValueArg`` as an ``option`` -- if it is there, regardless of its value, we are marking the includes as system.

Reply:
so we need to swtich to 
```cmake
  if( ${arg_TREAT_INCLUDES_AS_SYSTEM} )
  ...
  endif()
```

?

Reply:
Yes, I think so.